aws-java-sdk-s3/src/main/java/com/amazonaws/services/s3/transfer/internal/DownloadImpl.java

```diff
@@ -77,7 +74,7 @@ public synchronized ObjectMetadata getObjectMetadata() {
         if (s3Object != null) {
             return s3Object.getObjectMetadata();
         }
-        return objectMetadata;
+        return null;
```

Comment:
This will be a breaking change as customers who are getting object metadata before will get null now.